tritium-metrics/src/test/java/com/palantir/tritium/metrics/executor/TaggedMetricsScheduledExecutorServiceTest.java

```diff
@@ -0,0 +1,152 @@
+        assertThat(registry.meter(COMPLETED).getCount()).isEqualTo(0);
+        assertThat(registry.timer(DURATION).getCount()).isEqualTo(0);
+
+        finishLatch.countDown();
+        future.get();
+
+        assertThat(registry.meter(SUBMITTED).getCount()).isEqualTo(1);
+        assertThat(registry.counter(RUNNING).getCount()).isEqualTo(0);
+        assertThat(registry.meter(COMPLETED).getCount()).isEqualTo(1);
+        assertThat(registry.timer(DURATION).getCount()).isEqualTo(1);
+    }
+
+    @Test
+    public void testScheduledMetrics() throws Exception {
+        assertThat(registry.getMetrics())
+                .containsKeys(SCHEDULED_ONCE, SCHEDULED_REPETITIVELY);
+
+        assertThat(registry.meter(SCHEDULED_ONCE).getCount()).isEqualTo(0);
+        assertThat(registry.meter(SCHEDULED_REPETITIVELY).getCount()).isEqualTo(0);
+
+        executorService.schedule(() -> { }, 1L, TimeUnit.DAYS);
+
+        assertThat(registry.meter(SCHEDULED_ONCE).getCount()).isEqualTo(1);
+        assertThat(registry.meter(SCHEDULED_REPETITIVELY).getCount()).isEqualTo(0);
+
+        executorService.scheduleAtFixedRate(() -> { }, 1L, 1L, TimeUnit.DAYS);
+
+        assertThat(registry.meter(SCHEDULED_ONCE).getCount()).isEqualTo(1);
+        assertThat(registry.meter(SCHEDULED_REPETITIVELY).getCount()).isEqualTo(1);
+
+        executorService.scheduleWithFixedDelay(() -> { }, 1L, 1L, TimeUnit.DAYS);
+
+        assertThat(registry.meter(SCHEDULED_ONCE).getCount()).isEqualTo(1);
+        assertThat(registry.meter(SCHEDULED_REPETITIVELY).getCount()).isEqualTo(2);
+    }
+
+    @Test
+    public void testScheduledDurationMetrics() throws Exception {
+        assertThat(registry.getMetrics())
+                .containsKeys(SCHEDULED_OVERRAN, SCHEDULED_PERCENT_OF_PERIOD);
+
+        assertThat(registry.counter(SCHEDULED_OVERRAN).getCount()).isEqualTo(0);
+        assertThat(registry.histogram(SCHEDULED_PERCENT_OF_PERIOD).getCount()).isEqualTo(0);
+
+        Semaphore startSemaphore = new Semaphore(0);
+        Semaphore finishSemaphore = new Semaphore(1);
+        executorService.scheduleAtFixedRate(() -> {
+            startSemaphore.release();
+            finishSemaphore.acquireUninterruptibly();
+        }, 0L, 1L, TimeUnit.MILLISECONDS);
+
+        startSemaphore.acquire(2);
+
+        assertThat(registry.counter(SCHEDULED_OVERRAN).getCount()).isEqualTo(0);
+        assertThat(registry.histogram(SCHEDULED_PERCENT_OF_PERIOD).getCount()).isEqualTo(1);
+
+        TimeUnit.MILLISECONDS.sleep(2);
+        finishSemaphore.release();
+        startSemaphore.acquire();
+
+        assertThat(registry.counter(SCHEDULED_OVERRAN).getCount()).isEqualTo(1);
+        assertThat(registry.histogram(SCHEDULED_PERCENT_OF_PERIOD).getCount()).isEqualTo(2);
+    }
+
+    private static MetricName metricName(String metricName) {
+        return MetricName.builder()
+                .safeName(MetricRegistry.name("com/palantir/tritium/metrics/executor", metricName))
```

Comment:
```suggestion
                .safeName(MetricRegistry.name("scheduled-executor", metricName))
```

Reply:
I don't think we want this. The motivation to use common metric names with tags is to have a single metric to measure all executors. Splitting up the metric names again feels like a step backwards.